### DISCUSSION TITLE
Get data from database

### DISCUSSION THREAD

**Discussion by PanterSoft** [appwrite/appwrite#5660] [Q&A]
Hello,

I am not quite sure if I overread something but my question would be how I can extract data from the appwrite database after fetching it via the database.getDocument function.

<img width="484" alt="appwrite_response" src="https://github.com/appwrite/appwrite/assets/96704103/693bb6be-e991-471b-bf6f-76e6e46df1d5">

I want to extract the 258 Bytes of data. but somehow I don't get it how to do it.

I am using Swift 5 and the corresponding Appwrite Apple SDK (latest version)

I would really appreciate some help.

thanks in advance

**Comment by joeyouss** [comment]
Hey there,

I apologize for the delay in getting back to you. To retrieve a specific document from the database, you would need to use the document ID. By utilizing the reference mentioned in the documentation (https://appwrite.io/docs/client/databases?sdk=apple-default#databasesGetDocument), you can find detailed instructions on how to access the desired document.

If you have any further questions or need additional assistance, feel free to ask. I'm here to help!

**Comment by stnguyen90** [comment]
`getDocument()` returns a `Document` that has a `data`. You can see the source code for it here: https://github.com/appwrite/sdk-for-apple/blob/master/Sources/AppwriteModels/Document.swift#L66.

I don't know swift, but it looks like that `data` should be a map of some sort.

We also have some demo projects that might be helpful: https://github.com/appwrite?q=apple&type=all&language=&sort=
